test/langtools/tools/lib/toolbox/JavacTask.java

```diff
@@ -314,6 +314,26 @@ public String name() {
         return "javac";
     }
 
+    @Override
+    public Result run(Expect expect) {
+        int expectedExitCode = expect == Expect.SUCCESS ? 0 : 1;
+
+        return run(expect, (exitCode, testName) -> {
+            if (exitCode == 4) {
+                throw new TaskError("Task " + testName + " failed due to a javac crash "
+                    + "(exit code 4)");
+            }
+        });
+    }
+
+    @Override
+    public Result run(Expect expect, int exitCode) {
+        if (exitCode == 4) {
```

Comment:
I guess we never expect `javac` to crash, and should not be able to test that it does, right?

Reply:
I am not too worried about a test that would intentionally crash javac, and would need to test for that. It is not clear to me what would be the purpose of the test. And, assuming this need would indeed be rare, there are other ways to test for that. So, I opted for a more strict approach. OTOH, I don't have a too strong opinion on this, so I can re-allow exit code 4 here if desired.